src/Combinatorics/Graphs/functions.jl

```diff
@@ -1265,17 +1396,133 @@ Directed graph with 4 nodes and the following edges:
 ```
 """
 function graph_from_edges(::Type{T},
-                          edges::Vector{Vector{Int}},
-                          n_vertices::Int=-1) where {T <: Union{Directed, Undirected}}
+                          edges::Vector{S},
+                          n_vertices::Int=-1) where {T <: Union{Directed, Undirected}, S <: Union{Vector{Int}, NTuple{2, Int}}}
   return graph_from_edges(T, [Edge(e[1], e[2]) for e in edges], n_vertices)
 end
 
-function graph_from_edges(edges::Vector{Vector{Int}},
-                          n_vertices::Int=-1)
+function graph_from_edges(edges::Vector{T},
+                          n_vertices::Int=-1) where T <: Union{Vector{Int}, NTuple{2, Int}}
   return graph_from_edges(Undirected, [Edge(e[1], e[2]) for e in edges], n_vertices)
 end
 
+@doc raw"""
+     add_label!(G::Graph{T}, edge_labels::Union{Dict{Tuple{Int, Int}, Union{String, Int}}, Nothing}, vertex_labels::Union{Dict{Int, Union{String, Int}}, Nothing}=nothing; name::Symbol=:label) where {T <: Union{Directed, Undirected}}
+Given a graph `G` add a label to the edges and optinally to the vertixes with the given `name`
+
+```jldoctest
+julia> G = graph_from_edges(Directed, [[1, 2], [2, 3]])
+Directed graph with 3 nodes and the following edges:
+(1, 2)(2, 3)
+
+julia> add_label!(G, Dict((1, 2) => 1), nothing; name=:colour)
+Directed graph with 3 nodes and the following labelling(s)
+label: colour
+(1, 2) -> 1
+(2, 3) -> 0
+
+julia> G = graph_from_labelled_edges(Undirected, Dict((1, 2) => 1, (2, 3) => 2, (1, 3) => 1), nothing; name=:colour)
+Undirected graph with 3 nodes and the following labelling(s)
+label: colour
+(2, 1) -> 1
+(3, 1) -> 1
+(3, 2) -> 2
+
+julia> add_label!(G, nothing, Dict(1 => 1); name=:shading)
+Undirected graph with 3 nodes and the following labelling(s)
+label: colour
+(2, 1) -> 1
+(3, 1) -> 1
+(3, 2) -> 2
+label: shading
+1 -> 1
+2 -> 0
+3 -> 0
+
+```
+"""
+function add_label!(G::Graph{T},
+                    edge_labels::Dict{NTuple{2, Int}, S},
+                    vertex_labels::Dict{Int, U};
+                    name::Symbol=:label) where {S <: Union{Int, String}, U <: Union{Int, String}, T <: Union{Directed, Undirected}}
+  EM = EdgeMap{T, S}(pm_object(G))
+  NM = NodeMap{T, U}(pm_object(G))
+  set_attribute!(G, name, GraphMap(G, EM, NM))
+  for (k, v) in edge_labels
+    getproperty(G,name)[k] = v
+  end
+  for (k, v) in vertex_labels
+    @req k <= number_of_vertices(G) "Cannot label a vertex that is not in the graph, please set n_vertices"
```

Comment:
```suggestion
    @req k <= number_of_vertices(G) "Cannot label a vertex that is not in the graph"
```
Setting `n_vertices` only makes sense for a new graph, to modify the current graph one could use `add_vertices!`.